bridge-token-factory/src/lib.rs

```diff
@@ -110,6 +128,15 @@ pub trait ExtBridgeToken {
         memo: Option<String>,
         msg: String,
     ) -> PromiseOrValue<U128>;
+
+    fn set_metadata(
+        &mut self,
+        name: Option<String>,
+        symbol: Option<String>,
+        reference: Option<String>,
+        reference_hash: Option<Base64VecU8>,
+        decimals: Option<u8>,
+    );
```

Comment:
Need to add `icon: Option<String>,` field there for [NEP-148](https://nomicon.io/Standards/FungibleToken/Metadata.html) compliance